python/doc/theory/data_analysis/kolmogorov_test.rst

```diff
@@ -33,28 +33,45 @@ With a sample :math:`\left\{ x_1,x_2,\ldots,x_N \right\}`, the distance is estim
 
     \widehat{D}_N = \sup_{i=1 \ldots N}\left|F\left(x_i\right)-\frac{i-1}{N} ; \frac{i}{N}-F\left(x_i\right)\right|
 
-The probability distribution of the distance :math:`\widehat{D}_N` is
-asymptotically known (i.e. as the size of the sample tends to infinity).
-If *N* is sufficiently large, this means that for a probability
-:math:`\alpha` and a candidate distribution type, one can compute the
-threshold / critical value :math:`d_\alpha` such that:
-
-- if  :math:`\widehat{D}_N>d_{\alpha}`, we reject the candidate distribution
-  with a risk of error :math:`\alpha`,
-- if  :math:`\widehat{D}_N \leq d_{\alpha}`, the candidate distribution is
-  considered acceptable.
-
-Note that :math:`d_\alpha` does not depend on the candidate distribution
-*F* being tested, and the test is therefore relevant for any continuous
-distribution.
-
-An important notion is the so-called *p*-value of the test. This quantity is
-equal to the limit error probability :math:`\alpha_\textrm{lim}` under which
-the candidate distribution is rejected.
-Thus, the candidate distribution will be accepted if and only if
-:math:`\alpha_\textrm{lim}` is greater than the value :math:`\alpha` desired by
-the user.
-Note that the higher :math:`\alpha_\textrm{lim} - \alpha`, the more robust the decision.
+Assume that the sample is drawn from the candidate distribution. 
+By definition, the *p*-value of the test is the probability:
+
+.. math::
+
+    p = P(D > \widehat{D}_N)
+
+In the case where the fit is good, the value of :math:`\widehat{D}_N` is 
+small, which leads to a p-value closer to 1. 
+The candidate distribution will not be rejected if and only if 
+:math:`p` is larger than a given threshold probability. 
+In general, the threshold p-value is chosen to be 0.05:
+
+.. math::
+
+    p_{ref} = 0.05
+
+Based on the p-value, 
+- if  :math:`p<p_{ref}`, we reject the candidate distribution,
+- otherwise, the candidate distribution is not rejected.
+
+Two situations may occur in practice.
+- the parameters of the distribution under test are known,
+- the parameters of the distribution under test are estimated from data. 
+
+If the parameters of the distribution under test are known, 
+algorithms are available to directly compute 
+the distribution of :math:`\widehat{D}_N` both for *N* large 
+(asymptotic distribution) or for *N* finite. 
+This is because the distribution of :math:`\widehat{D}_N` does 
+not depend on the candidate distribution. 
+
+If the parameters of the distribution under test are estimated 
+from data, the :math:`\widehat{D}_N` statistic is generally smaller, because 
+the parameters of the distribution have been computed from the 
+data. 
+In general, the distribution of :math:`\widehat{D}_N` is not known 
```

Comment:
I would say that the distribution of D_n is lightly differ comparing to D distribution and we should involve Bootstrap methods to better estimate the pvalue

Reply:
I am not sure I clearly understand this specific comment. I suppose that you suggest to describe the algorithm which presumably uses bootstrap, but this is not true because the algorithm does not involve bootstrap. Please look at:

https://github.com/openturns/openturns/blob/master/lib/src/Uncertainty/StatTests/FittingTest.cxx#L272

The algorithm is the following.
* Estimate the parameters of the distribution A based on the sample S. 
* Compute the KS distance D between A and S.
* Get the size N of the sample.
* For i in sampleSize (e.g. sampleSize = 1000)
  * Generate the sample T from distribution A and size N.
  * Estimate the parameters of the distribution B based on the sample T.
  * Compute and store the KS distance between T and B.
* Compute the p-value of the KS distance D within the KS distance sample.

There is no bootstrap here, in the sense that we generate samples from a distribution, not from the sample itself. I seems to me that this algorithm does not fit in the current documentation, but this is just a choice.

Reply:
es, excepting mistakes in the written lines. Btw, global idea is to highlight the difference between distribution's parameters in case these are to be estiamted or are known